custom-assembly-report-generator/generate_custom_assembly_report.py

```diff
@@ -0,0 +1,184 @@
+    urllib.urlretrieve(assembly_report_url, assembly_report_path)
+    urllib.urlcleanup()
+    return assembly_report_path
+
+
+def insert_absent_genbank_accessions_in_assembly_report(species_db_info, assembly_report_dataframe,
+                                                        genbank_equivalents_dataframe):
+    absent_refseq_accessions = \
+        get_absent_refseq_accessions(species_db_info, set(assembly_report_dataframe["RefSeq-Accn"]))
+    if len(absent_refseq_accessions) == 0:
+        print("INFO: No GenBank accessions are absent in the assembly report for " + species_db_info["database_name"])
+    new_assembly_report_entry_index = len(assembly_report_dataframe)
+    for refseq_accession, chromosome, assembly_unit, sequence_length in absent_refseq_accessions:
+        equivalent_genbank_accession = \
+            genbank_equivalents_dataframe[genbank_equivalents_dataframe["rs_acc"] == refseq_accession]["gb_acc"]
+        if len(equivalent_genbank_accession) == 0:
+            print("WARN: Could not find equivalent GenBank accession for RefSeq accession: " + refseq_accession)
+        else:
+            equivalent_genbank_accession = equivalent_genbank_accession.values[0]
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "Sequence-Role"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "Assigned-Molecule"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "Assigned-Molecule-Location/Type"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "GenBank-Accn"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "Relationship"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "RefSeq-Accn"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "Assembly-Unit"], \
+            assembly_report_dataframe.loc[new_assembly_report_entry_index, "Sequence-Length"] \
+            = "assembled-molecule", \
+              chromosome, \
+              "Chromosome", \
+              equivalent_genbank_accession, \
+              "=", \
+              refseq_accession, \
+              assembly_unit, \
+              sequence_length
+            new_assembly_report_entry_index += 1
+    return assembly_report_dataframe
+
+
+def update_non_equivalent_genbank_accessions_in_assembly_report(assembly_report_dataframe,
+                                                                genbank_equivalents_dataframe):
+    refseq_accessions_with_non_equivalent_genbank_accessions = \
+        get_refseq_accessions_without_equivalent_genbank_accessions(assembly_report_dataframe)
+    if len(refseq_accessions_with_non_equivalent_genbank_accessions) == 0:
+        print("INFO: No entries were found in the assembly report with non-equivalent GenBank accessions")
+    for accession in refseq_accessions_with_non_equivalent_genbank_accessions:

```

Comment:
Blank line to increase readability.

```suggestion

    for accession in refseq_accessions_with_non_equivalent_genbank_accessions:
```